lib/ansible/plugins/lookup/sops.py

```diff
@@ -0,0 +1,162 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError, AnsibleParserError, AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_text, to_native
+
+from ansible.utils.display import Display
+display = Display()
+
+from subprocess import Popen, PIPE
+
+DOCUMENTATION = """
+    lookup: sops
+    author: Edoardo Tenani <e.tenani@arduino.cc>
+    version_added: "2.8"
+    short_description: Read sops encrypted file contents
+    description:
+        - This lookup returns the contents from a file on the Ansible controller's file system.
+        - This lookup requires the sops executable to be available in the controller PATH.
+    options:
+        _terms:
+            description: path(s) of files to read
+            required: True
+        print:
+            description: print the decrypted content
+            type: bool
+            required: False
+            default: False
+    notes:
+        - this lookup does not understand 'globing' - use the fileglob lookup instead.
+"""
+
+EXAMPLES = """
+tasks:
+  - name : output secrets to screen (BAD IDEA)
+    debug:
+        msg: "Content: {{lookup('sops', item)}}"
+    with_items:
+        - sops-encrypted-file.enc.yaml
+
+  - name: add ssh private key
+    copy:
+        content: "{{lookup('sops', user + '-id_rsa')}}"
+        dest: /home/{{user}}/.ssh/id_rsa
+        owner: "{{user}}"
+        group: "{{user}}"
+        mode: 0600
+"""
+
+RETURN = """
+    _raw:
+        description: decrypted file content
+"""
+
+# From https://github.com/mozilla/sops/blob/master/cmd/sops/codes/codes.go
+# Should be manually updated
+sops_error_codes = {
+    1: "ErrorGeneric",
+    2: "CouldNotReadInputFile",
+    3: "CouldNotWriteOutputFile",
+    4: "ErrorDumpingTree",
+    5: "ErrorReadingConfig",
+    6: "ErrorInvalidKMSEncryptionContextFormat",
+    7: "ErrorInvalidSetFormat",
+    8: "ErrorConflictingParameters",
+    21: "ErrorEncryptingMac",
+    23: "ErrorEncryptingTree",
+    24: "ErrorDecryptingMac",
+    25: "ErrorDecryptingTree",
+    49: "CannotChangeKeysFromNonExistentFile",
+    51: "MacMismatch",
+    52: "MacNotFound",
+    61: "ConfigFileNotFound",
+    85: "KeyboardInterrupt",
+    91: "InvalidTreePathFormat",
+    100: "NoFileSpecified",
+    128: "CouldNotRetrieveKey",
+    111: "NoEncryptionKeyFound",
+    200: "FileHasNotBeenModified",
+    201: "NoEditorFound",
+    202: "FailedToCompareVersions",
+    203: "FileAlreadyEncrypted"
+}
+
+
+class Sops():
+    @staticmethod
+    def decrypt(encrypted_file):
+        # Run sops directly, python module is deprecated
+        command = ["sops", "--decrypt", encrypted_file]
+        process = Popen(command, stdout=PIPE, stderr=PIPE)
```

Comment:
Would this be better? `process = Popen(command, stdout=PIPE, stderr=PIPE, encoding='utf8')`

Reply:
This was due to an environment issue on my end, so disregard.